### PR TITLE
Podcast feature

### PR DESCRIPTION
**Pull request by Ivan12273**
Context: Adding a new feature for listen to podcast from iTunes API.

Hello! I am working on the issue [#966](https://github.com/nukeop/nuclear/issues/966) I added the next changes:

- **rest/iTunesPodcast.ts:** four functions for obtaining the information from the iTunes API.
- **stream/iTunesPodcastPlugin.ts:** search and getAlternateStream (I needed to do two request for each one, one for obtaining the id of the podcast and the other for the episode, the iTunes API has its limitations for searching)
- **meta/iTunespodcast.ts:** for obtaining the details of the podcast and the episodes. I did it in artist in order to test what I have, I was trying to do it in album but I noticed that the album tab works different that the artist tab.

I solved my problem I mentioned with the id.
I was thinking of creating a new podcast tab or re-use the album tab, but still not sure how the album components work, I would appreciate if someone can help guide me through the parts that I need to modify to add a new tab as album :)

### CODE REVIEW

**Comment by nukeop** [comment]
> I was thinking of creating a new podcast tab or re-use the album tab, but still not sure how the album components work, I would appreciate if someone can help guide me through the parts that I need to modify to add a new tab as album :)

I think it would be good to just use the same album grid as albums use.

Search results are provided by this container: https://github.com/nukeop/nuclear/blob/master/packages/app/app/containers/SearchResultsContainer/index.js
And they're passed to this component: https://github.com/nukeop/nuclear/blob/master/packages/app/app/components/SearchResults/index.js

If you format the results correctly, you will be able to add an item under `panes()`, using `renderPane`.
If you're able, please also add a test for this here: https://github.com/nukeop/nuclear/tree/master/packages/app/app/containers/SearchResultsContainer

**Comment by Ivan12273** [comment]
Hello, I made new changes in:

- /components/SearchResults/index.js
- /containers/SearchResultsContainer/index.js

And other files in order to add a new tab for podcast, I re-use the album grid and other album functions for display the podcasts.

Also I added a new abstract function in metaProvider.

image

image

It worked fine, are these changes correct?

File: packages/app/app/components/AlbumView/index.tsx [hunk 1]
```diff
@@ -138,7 +138,7 @@ export const AlbumView: React.FC<AlbumViewProps> = ({
                   <th className={cx(styles.center, styles.position)}>
                     <Icon name='hashtag' />
                   </th>
-                  <th className={styles.left}>Song</th>
+                  <th className={styles.left}>Tracks</th>
```
Comment (HaiDang666) [inline packages/app/app/components/AlbumView/index.tsx:141]: can you use locale in stead of this hard string

File: packages/core/src/plugins/meta/audius.ts [hunk 2]
```diff
@@ -128,10 +129,13 @@ class AudiusMetaProvider extends MetaProvider {
     throw new Error('Method not implemented.');
   }
 
-
   async fetchAlbumDetailsByName(): Promise<AlbumDetails> {
     throw new Error('Method not implemented.');
   }
+
+  searchForPodcast(): Promise<SearchResultsPodcast[]> {
```
Comment (HaiDang666) [inline packages/core/src/plugins/meta/audius.ts:136]: @nukeop is it possible to make an interface for `MetaProvider` to implement default method to return empty value. So we don't need to do like this in very meta provider when we come ups with new method.

**Review by HaiDang666 — CHANGES_REQUESTED** [review]

File: packages/app/app/components/AlbumView/index.tsx (hunk 1 above)
Comment (Ivan12273) [reply packages/app/app/components/AlbumView/index.tsx:141]: Sorry, what did you mean with locale?

Comment (Ivan12273) [reply packages/app/app/components/AlbumView/index.tsx:141]: I changed it to how it was originally

Comment (nukeop) [reply packages/app/app/components/AlbumView/index.tsx:141]: These need to be translated according to the selected language, using `react-i18next`.

File: packages/core/src/plugins/meta/audius.ts (hunk 2 above)
Comment (nukeop) [reply packages/core/src/plugins/meta/audius.ts:136]: @Ivan12273 You can add this method to this abstract class with a default implementation: https://github.com/nukeop/nuclear/blob/6a99b3f85042f170de840d3e9b9381d8cd5fd84b/packages/core/src/plugins/metaProvider.ts

This way you can remove it from child providers.

**Comment by nukeop** [comment]
I agree with @HaiDang666 's comments. Very good work too.

File: packages/app/app/components/AlbumView/index.tsx (hunk 1 above)
Comment (HaiDang666) [reply packages/app/app/components/AlbumView/index.tsx:141]: you can see in the same file that:
```nodejs
const { t } = useTranslation('album');
...
<PopupButton
    onClick={addAlbumToDownloads}
    ariaLabel={t('download')}
    icon='download'
    label={t('download')}
  />
```
and the `download` key is defined in `packages/i18n/src/locales/en.json` as:
```js
{
  ...
  album: {
     ...
     download: 'Download'
  }
}
```

Comment (Ivan12273) [reply packages/app/app/components/AlbumView/index.tsx]: Hi! I changed it again using `react-i18next` I was only able to translate to Spanish, the others are in English :)

Comment (HaiDang666) [reply packages/app/app/components/AlbumView/index.tsx]: That good 😌

File: packages/core/src/plugins/meta/audius.ts (hunk 2 above)
Comment (Ivan12273) [reply packages/core/src/plugins/meta/audius.ts]: I added the method with the default implementation and removed it from the others classes. I needed to add` // eslint-disable-next-line @typescript-eslint/no-unused-vars` for the unused var :)

**Comment by HaiDang666** [comment]
I found test snapshot fail and this
Screen Shot 2021-06-20 at 16 39 39

**Comment by Ivan12273** [comment]
> I found test snapshot fail and this
> Screen Shot 2021-06-20 at 16 39 39
> 
> Screen Shot 2021-06-20 at 16 43 50

Hi, I corrected it, it was because I changed the text haha.

image

The first one is because of the file name, in my local project I have it different, I already changed it in the repo too :)

**Review by HaiDang666 — APPROVED** [review]
Good work, just wait for @nukeop.

**Comment by nukeop** [comment]
Thanks, this is a really good contribution.

**Pull request merged**